contracts/profit/src/contract.rs

```diff
@@ -61,28 +63,91 @@ pub fn migrate(deps: DepsMut<'_>, _env: Env, _msg: MigrateMsg) -> ContractResult
 ) -> ContractResult<CwResponse> {
     match msg {
         ExecuteMsg::TimeAlarm {} => {
+            SingleUserAccess::load(deps.storage, crate::access_control::TIMEALARMS_NAMESPACE)?
+                .check_access(&info.sender)?;
+
             let alarm_recepient = env.contract.address.clone();
-            try_transfer(deps.as_ref(), env, info).and_then(|resp| {
+
+            try_time_alarm(deps.branch(), env).and_then(|resp| {
                 response::response_with_messages::<_, _, ContractError>(&alarm_recepient, resp)
             })
         }
+        ExecuteMsg::Config { cadence_hours } => {
+            SingleUserAccess::check_owner_access::<ContractError>(deps.storage, &info.sender)?;
+
+            State::load(deps.storage)?
+                .try_update_config(cadence_hours)?
+                .store(deps.storage)?;
+
+            Ok(response::empty_response())
+        }
     }
 }
 
 #[cfg_attr(feature = "contract-with-bindings", entry_point)]
-pub fn sudo(deps: DepsMut<'_>, _env: Env, msg: SudoMsg) -> ContractResult<CwResponse> {
-    match msg {
-        SudoMsg::Config { cadence_hours } => {
-            Profit::try_config(deps.storage, cadence_hours).map(|()| response::empty_response())
+pub fn sudo(deps: DepsMut<'_>, env: Env, msg: NeutronSudoMsg) -> ContractResult<CwResponse> {
+    let state: State = State::load(deps.storage)?;
+
+    let DexResponse::<State> {
+        response,
+        next_state,
+    } = match msg {
+        NeutronSudoMsg::Response { data, .. } => {
+            match state.on_response(data, deps.as_ref(), env) {
+                DexResult::Continue(result) => result?,
+                DexResult::Finished(response) => response,
+            }
         }
-    }
+        NeutronSudoMsg::Error { .. } => state.on_error(deps.as_ref(), env)?,
+        NeutronSudoMsg::Timeout { .. } => state.on_timeout(deps.as_ref(), env)?,
+        NeutronSudoMsg::OpenAck {
+            channel_id,
+            counterparty_channel_id,
+            counterparty_version,
+            ..
+        } => state.confirm_open(
+            deps.as_ref(),
+            env,
+            Ics20Channel {
+                local_endpoint: channel_id,
+                remote_endpoint: counterparty_channel_id,
+            },
+            counterparty_version,
+        )?,
+        NeutronSudoMsg::TxQueryResult { .. } => {
+            unimplemented!()
+        }
+        NeutronSudoMsg::KVQueryResult { .. } => {
+            unimplemented!()
+        }
+    };
+
+    next_state.store(deps.storage)?;
+
+    Ok(response::response_only_messages(response))
+}
+
+fn try_time_alarm(deps: DepsMut<'_>, env: Env) -> ContractResult<MessageResponse> {
+    let state: State = State::load(deps.storage)?;
+
+    let DexResponse::<State> {
+        response,
+        next_state,
+    } = match state.on_time_alarm(deps.as_ref(), env) {
+        DexResult::Continue(result) => result?,
+        DexResult::Finished(response) => response,
+    };
```

Comment:
this is already implemented, Look at packages/dex/src/response.rs for `impl<H, Err> From<Result<H>> for StdResult<Response<H>, Err>`. Then use `Into::into` instead.

Reply:
Actually not implemented for this case as it's specialized for cases where `Handler::SwapResult` is a `std::result::Result`, not `dex::response::Response`.

Implemented a new method for that case, though.